File: app/src/main/java/com/example/foundlah/ConvoAdapter.java
package com.example.foundlah;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class ConvoAdapter extends RecyclerView.Adapter<ConvoAdapter.ConvoViewHolder>{
    private List<Convo> convoList;
    private OnUserClickListener onUserClickListener;

    public interface OnUserClickListener {
        void onUserClick(Convo convo);
    }

    public ConvoAdapter(List<Convo> convoList, OnUserClickListener onUserClickListener) {
        this.convoList = convoList;
        this.onUserClickListener = onUserClickListener;
    }

    public ConvoViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_convo, parent, false);
        return new ConvoViewHolder(view);
    }

    public void onBindViewHolder(@NonNull ConvoViewHolder holder, int position) {
        Convo convo = convoList.get(position);
        holder.user1.setText(convo.getUser1());
        holder.user2.setText(convo.getUser2());

        Log.d("ConvoAdapter", "Binding" + convo.getUser1() + "&" + convo.getUser2());
        holder.itemView.setOnClickListener(v -> onUserClickListener.onUserClick(convo));
    }

    @Override
    public int getItemCount() {
        return convoList.size();
    }

    public static class ConvoViewHolder extends RecyclerView.ViewHolder {
        TextView user1, user2;

        public ConvoViewHolder(@NonNull View itemView) {
            super(itemView);
            user1 = itemView.findViewById(R.id.user1);
            user2 = itemView.findViewById(R.id.user2);
        }
    }
}

File: app/src/main/java/com/example/foundlah/ConvoAdapter.kt
package com.example.foundlah

import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.foundlah.ConvoAdapter.ConvoViewHolder

class ConvoAdapter(
    private val convoList: List<Convo>,
    private val onUserClickListener: OnUserClickListener
) :
    RecyclerView.Adapter<ConvoViewHolder>() {
    fun interface OnUserClickListener {
        fun onUserClick(convo: Convo?)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ConvoViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_convo, parent, false)
        return ConvoViewHolder(view)
    }

    override fun onBindViewHolder(holder: ConvoViewHolder, position: Int) {
        val convo = convoList[position]
        holder.user1.text = convo.user1
        holder.user2.text = convo.user2

        Log.d("ConvoAdapter", "Binding" + convo.user1 + "&" + convo.user2)
        holder.itemView.setOnClickListener { v: View? ->
            onUserClickListener.onUserClick(
                convo
            )
        }
    }

    override fun getItemCount(): Int {
        return convoList.size
    }

    class ConvoViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var user1: TextView =
            itemView.findViewById(R.id.user1)
        var user2: TextView =
            itemView.findViewById(R.id.user2)
    }
}
